frontend/src/pages/CourseSelector/CourseDescription/PrerequisiteTree/PrerequisiteTree.jsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import G6 from "@antv/g6";
+import Spinner from "components/Spinner";
+import axiosRequest from "config/axios";
+import { addTab } from "reducers/courseTabsSlice";
+import S from "./styles";
+
+const PrerequisiteTree = ({ currCourse }) => {
+  const [loading, setLoading] = useState(true);
+  const [graph, setGraph] = useState(null);
+  const [coursesPathTo, setCoursesPathTo] = useState([]);
+  const [coursesPathFrom, setCoursesPathFrom] = useState([]);
+  const dispatch = useDispatch();
+  const ref = useRef(null);
+  const { degree, planner } = useSelector((state) => state);
+
+  /* CONSTANTS */
+  const relationship = {
+    PREREQ: "prereq",
+    UNLOCKS: "unlocks",
+  }
+
+  /* GRAPH IMPLEMENTATION */
+  const getHeight = () => {
+    let maxCourseGroupNum = Math.max(coursesPathFrom.length, coursesPathTo.length);
+    maxCourseGroupNum = maxCourseGroupNum === 0 ? 1 : maxCourseGroupNum;
+    // take node height as 40
+    return 40 * maxCourseGroupNum;
```

Comment:
This is an estimation of the node height?

Reply:
tis, i tried setting node height in defaultNode but it wasn't working rip